File: src/main/java/com/intellij/tapestry/intellij/toolwindow/nodes/UsagesNode.java
package com.intellij.tapestry.intellij.toolwindow.nodes;

import com.intellij.tapestry.core.TapestryProject;
import com.intellij.tapestry.core.model.presentation.PresentationLibraryElement;

import javax.swing.tree.DefaultMutableTreeNode;

public class UsagesNode extends DefaultMutableTreeNode {

    public UsagesNode(Object userObject) {
        super(userObject);

        PresentationLibraryElement element = (PresentationLibraryElement) userObject;
        for (TapestryProject.Usage usage : element.getProject().findUsages(element)) {
            add(new UsageNode(usage.user(), usage.kind()));
        }
    }

    public String toString() {
        return "Used By";
    }
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/intellij/toolwindow/nodes/UsagesNode.kt
package com.github.rar91279.plugin.tapestry.intellij.toolwindow.nodes

import com.github.rar91279.plugin.tapestry.core.model.presentation.PresentationLibraryElement
import javax.swing.tree.DefaultMutableTreeNode

class UsagesNode(element: PresentationLibraryElement) : DefaultMutableTreeNode(element) {

    init {
        for (usage in element.project.findUsages(element)) add(UsageNode(usage.user, usage.kind))
    }

    override fun toString() = "Used By"
}
